[fix] at::view

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#23452 [fix] at::view**

accidently calls clone, but what we want is creating an empty tensor and set storage.

Differential Revision: [D16442756](https://our.internmc.facebook.com/intern/diff/D16442756/)

jerryzh168 committed:
[fix] at::view

accidently calls clone, but what we want is creating an empty tensor and set storage.

Differential Revision: [D16442756](https://our.internmc.facebook.com/intern/diff/D16442756/)

jerryzh168 committed:
Update on "[fix] at::view"


accidently calls clone, but what we want is creating an empty tensor and set storage.

Differential Revision: [D16442756](https://our.internmc.facebook.com/intern/diff/D16442756/)

zafartahirov commented on aten/src/ATen/native/TensorShape.cpp:
```diff
@@ -898,20 +898,38 @@ Tensor numpy_T(const Tensor &self) {
+Tensor alias(const Tensor& self) {
+  Tensor self_;
+  if (self.is_quantized()) {
+    TORCH_CHECK(self.qscheme() == kPerTensorAffine,
```
Should we allow `kPerTensorSymmetric` in here as well?

jerryzh168 replied:
we don't support that right now

gchanan replied:
nit: shouldn't the QTensorImpl constructor check that then?

jerryzh168 replied:
I think we have per channel quantizer right now. looking again, the following code actually works for whatever tensor we have, so we can remove this check.

z-a-f reviewed (approved):
LGTM

jerryzh168 committed:
Update on "[fix] at::view"


accidently calls clone, but what we want is creating an empty tensor and set storage.

Differential Revision: [D16442756](https://our.internmc.facebook.com/intern/diff/D16442756/)

jerryzh168 committed:
Update on "[fix] at::view"


accidently calls clone, but what we want is creating an empty tensor and set storage.

Differential Revision: [D16442756](https://our.internmc.facebook.com/intern/diff/D16442756/)

jerryzh168 commented:
@pytorchbot retest this please

jerryzh168 committed:
Update on "[fix] at::view"


accidently calls clone, but what we want is creating an empty tensor and set storage.

Differential Revision: [D16442756](https://our.internmc.facebook.com/intern/diff/D16442756/)

jerryzh168 commented:
@pytorchbot retest this please

jerryzh168 commented:
@pytorchb retest this please

gchanan commented on aten/src/ATen/test/basic.cpp:
```diff
@@ -283,6 +283,14 @@ void TestNegativeDim(DeprecatedTypeProperties& type) {
   ASSERT_ANY_THROW(tensor.reshape({-5, -5}));
 }
 
+void TestView(DeprecatedTypeProperties& type) {
```
nit: would be good to comment why you need this test; that this tests the *Tensor* view path, not the *Variable* view path, which has different semantics (you could link this PR for more details).

jerryzh168 committed:
Update on "[fix] at::view"


accidently calls clone, but what we want is creating an empty tensor and set storage.

Differential Revision: [D16442756](https://our.internmc.facebook.com/intern/diff/D16442756/)

jerryzh168 committed:
Update on "[fix] at::view"


accidently calls clone, but what we want is creating an empty tensor and set storage.

Differential Revision: [D16442756](https://our.internmc.facebook.com/intern/diff/D16442756/)

jerryzh168 committed:
Update on "[fix] at::view"


accidently calls clone, but what we want is creating an empty tensor and set storage.

Differential Revision: [D16442756](https://our.internmc.facebook.com/intern/diff/D16442756/)

jerryzh168 committed:
Update on "[fix] at::view"


accidently calls clone, but what we want is creating an empty tensor and set storage.

Differential Revision: [D16442756](https://our.internmc.facebook.com/intern/diff/D16442756/)
